### DISCUSSION TITLE
Weighted network

### DISCUSSION THREAD

**Discussion by peaceGui** [JoelForamitti/agentpy#41] [Q&A]
Hey,
I'm trying to use a weighted network to model disease spread, however, I can't find any relevant examples.
for example， I am using the code below but I am getting an error：
> TypeError: getattr(): attribute name must be string

```
class Person(ap.Agent):

    def setup(self):
        """ Initialize a new variable at agent creation. """
        self.condition = 0  # Susceptible = 0, Infected = 1, Recovered = 2

    def being_sick(self):
        """ Spread disease to peers in the network. """
        rng = self.model.random
        for n in self.network.neighbors(self):
            if n.condition == 0 and self.p.infection_chance* self.network[self][n]['weight']> rng.random():
                n.condition = 1  # Infect susceptible peer
        if self.p.recovery_chance > rng.random():
            self.condition = 2  # Recover from infection
```

here, I use this code `self.network[self][n]['weight']`  to get the weight on this edge (this node to its neighbor n), but it doesn't work, how can I get the weight?

Thanks in advance for any advice!

**Comment by jofmi** [comment]
Hi! Sorry for the late answer. Can you show the code where you define the weight attribute?
